src/FlightMap/Widgets/BatteryVisual.qml

```diff
@@ -0,0 +1,117 @@
+            color:           getColorForPercentage(batteryContainer.percentageRemaining)
+            opacity:         0.8
+        }
+
+        Rectangle {
+            id:                   bingoLimit1Bar
+            anchors.left:         parent.left
+            anchors.right:        parent.right
+            anchors.bottom:       parent.bottom
+            anchors.bottomMargin: parent.batCapacity ? parent.calculateBottomMargin(1 - (_aviantSettings.preDeliveryBingoLimit.rawValue / parent.batCapacity.rawValue)) : 0
+            anchors.leftMargin:   parent.margin
+            anchors.rightMargin:  parent.margin
+            height:               ScreenTools.defaultFontPixelHeight / 6
+            visible:              parent.batCapacity !== null
+        }
+
+        Rectangle {
+            id:                   bingoLimit2Bar
+            anchors.left:         parent.left
+            anchors.right:        parent.right
+            anchors.bottom:       parent.bottom
+            anchors.bottomMargin: parent.batCapacity ? parent.calculateBottomMargin(1 - (_aviantSettings.postDeliveryBingoLimit.rawValue / parent.batCapacity.rawValue)) : 0
+            anchors.leftMargin:   parent.margin
+            anchors.rightMargin:  parent.margin
+            height:               ScreenTools.defaultFontPixelHeight / 6
+            visible:              batteryContainer.batCapacity !== null
+        }
+
+        Rectangle {
+            id:                   criticalThresholdBar
+            anchors.left:         parent.left
+            anchors.right:        parent.right
+            anchors.bottom:       parent.bottom
+            anchors.bottomMargin: batteryContainer.batCriticalThreshold ? parent.calculateBottomMargin(batteryContainer.batCriticalThreshold.rawValue) : 0
+            anchors.leftMargin:   parent.margin
+            anchors.rightMargin:  parent.margin
+            height:               ScreenTools.defaultFontPixelHeight / 6
+            color:                "#DD6504"
+            visible:              batteryContainer.batCriticalThreshold !== null
+        }
+
+        Rectangle {
+            id:                   emergencyThresholdBar
+            anchors.left:         parent.left
+            anchors.right:        parent.right
+            anchors.bottom:       parent.bottom
+            anchors.bottomMargin: batteryContainer.batEmergencyThreshold ? parent.calculateBottomMargin(batteryContainer.batEmergencyThreshold.rawValue) : 0
+            anchors.leftMargin:   parent.margin
+            anchors.rightMargin:  parent.margin
+            height:               ScreenTools.defaultFontPixelHeight / 6
+            color:                qgcPal.colorRed
+            visible:              batteryContainer.batEmergencyThreshold !== null
+        }
+    }
+
+    function getColorForPercentage(percent) {
+        if (percent <= 10) return qgcPal.colorRed
+        if (percent <= 25) return qgcPal.colorOrange
```

Comment:
Hard coded rules of 10% and 25% will be confusing to pilots because they will wonder what it means. Expected behavior would be to get color from battery failsafe limits, but having a constant color green is OK too

Reply:
Changed to constant green